### DISCUSSION TITLE
Branches are not merged even with automerge set to true

### DISCUSSION THREAD

**Discussion by robg-temp** [renovatebot/renovate#9425] [Request Help]
We are planning to use Renovate to **only update internal libraries**, which has proven to be quite tricky to configure.

It is configured as a jenkins job that runs every 15 minutes that scans about GitHub 10 projects.  The library matching seems to be working, but automerging never happens, even with "automerge" set to _true_.

According to Renovate, Automerge is enabled:

![image](https://user-images.githubusercontent.com/80468671/113869210-72a4ab80-977e-11eb-885e-d69c468f8a15.png)

There are really 2 issues:

1. A PR is still created even though the "automergeType" is set to "branch".
2. The PR is never merged, even after over a day.

Here's the central config.js file:

```
module.exports = {
  "baseBranches": ["master", "Sprint-150", "Release-152"],
  "branchPrefix": "renovate/",
  "onboarding": true,
  "onboardingConfig": {
    "extends": [ "config:base" ]
  },
  "platform": "github",
  "rangeStrategy": "bump",
  "automerge": true,
  "automergeType": "branch",
  "packageRules": [
    {
      "matchPackagePatterns": ["*"],
      "excludePackagePatterns": ["^@acme"],
      "enabled": false
    },
    {
      "separateMajorMinor": true,
      "major": {
        "enabled": false
      }
    },
    {
      "matchBaseBranches": ["Sprint-150", "Release-152"],
      "matchPackagePatterns": ["common-web-assets"],
      "separateMinorPatch": true,
      "minor": {
        "enabled": false
      }
    }
  ]
};
```

In case you're curious, our rules are:
On _master_ All packages starting with "@acme" should be updated (minor and patch).
On _release_ and _sprint_ branches, only the "common" package should be updated (patch only).

Any help would be much appreciated, as I have been struggling with this configuration for weeks!

Thanks!

Rob

**Comment by rarkins** [comment]
Please capture logs at debug level (set LOG_LEVEL=debug in env):
- On a run where you expect automerge to work, and/or a PR is created when you didn't expect one
- Look for the branch name in the logs + any mentions of automerge

**Reply by robg-temp** [reply]
Thanks so much for the quick reply.  Doing it now....

**Reply by viceice** [reply]
I think some github branch protections are blocking automerge, should be visible in debug logs

**Reply by robg-temp** [reply]
I think that you may be right.  We do have protection on master!

**Reply by robg-temp** [reply]
I'll see if there is something in the logs about why renovate is still making PRs.

**Reply by robg-temp** [reply]
I think I read somewhere in the docs that Renovate can be configured to approve its own PRs.  Maybe I can do that?

**Reply by robg-temp** [reply]
I am seeing something in the logs about automerge:

```
DEBUG: Checking if we can automerge branch (repository=Acme/ti, branch=renovate/Sprint-152-pin-dependencies)
DEBUG: mergeStatus=no automerge (repository=Acme/ti, branch=renovate/Sprint-152-pin-dependencies)
```

**Reply by robg-temp** [reply]
I found the bit about auto-approving PRs:

> If you are running the hosted WhiteSource Renovate App on github.com, you can also install the helper apps renovate-approve and renovate-approve-2 and they will mark all automerging Pull Requests by Renovate as approved. These approval helper apps are only available for GitHub.

**Reply by viceice** [reply]
https://github.com/apps/renovate-approve

**Reply by robg-temp** [reply]
Yeah, that's one of them. Thanks!

**Reply by rarkins** [reply]
If you have configured that PRs need approving then that in most cases would block branch automerge

**Reply by robg-temp** [reply]
That explains that problem.

**Reply by robg-temp** [reply]
Thanks for all your help on this!

**Reply by robg-temp** [reply]
I'm not sure if I should create a new discussion for this, but the automerging seems to be a bit haphazard.  Some projects have automerge enabled, while others do not.  **Any ideas why that is?**  The automerge is configured at the root level.

![image](https://user-images.githubusercontent.com/80468671/113888712-97a21a00-9790-11eb-9a26-57bade4e5fd0.png)

**Reply by rarkins** [reply]
Please create a public reproduction repo if you need anyone to dig into it.
